src/base/StringView.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 1996-2019 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS string_views for details.
+ */
+
+#ifndef SQUID_BASE_STRING_VIEW_H
+#define SQUID_BASE_STRING_VIEW_H
+
+#include <iosfwd>
+
+// TODO: Replace with std::string_view after switching to C+17
+/// a reference to portion of a character sequence
+/// 0-termination is neither assumed nor honored -- NUL is not treated specially
```

Comment:
In other words replicating SBuf functionality with c-strings. I see no need for this to be added to Squid - rather the opposite it is quite dangerous for us to do so. The performance issues this "avoids" were explicitly added to SBuf because we cannot accept their absence in string handling code.

Reply:
I stand by my objection to this dangerous class being added.

When the old StringArea code was dropped we left MemBuf as the object able to meet code needs which require access to unterminated char* in ways that we are unwilling to make SBuf support.

MemBuf X;
 X.buf = start;
 X.size = len;
 X.stolen = 1;

Is the equivalent to:
 StringView X(start,len)

@rousskov: If you want to make it formal I am open to adding a method or constructor to MemBuf that initializes it as a stolen memory area.

Reply:
It is supposed that we are trying to drop MemBuf from Squid or at least limit its use.

Moreover I can not understand why a dedicated class like the StringView where it is clear what it does is more dangerous than using the MemBuf which actually is a string class.
It is very possible to be confused with MemBuf and continue using it on a memory block which actually is freed or it does not exist any more.
StringView does what its name says: provides a view of an existing memory area.

I would be positive to add support for const SBuf with similar functionality with the MemBuf/MemBuf.stolen but with the precondition that this is going to be used with static strings only, eg:
```
const char *aString = "Hello String";
const SBuf aSBufString(aString);
```
But definitely no for cases like this. For cases like this we need something similar to StringView.

If the StringView is not acceptable my opinion is to revert all the related code to the state before commit 4763743.  And wait the HttpHeaders going to use SBuf instead of MemBuf, and then re-implement StrList API using completely different logic.

Reply:
The StringView class is removed with the commit 4365d547

Someone can improve performance (and do one step on moving from String to SBuf for HttpHeaders class) :
- Write a variant of strListGetItem which gets as argument an SBuf. It can return SBuf instead of "const char*". It can be a StrListIterator member and then mark existing strListGetItem as deprecated. 
- Make StrList accept and parse an SBuf instead of a String
- Add a method "bool getList(Http::HdrType id, SBuf &) const"

Although I had some thoughts to start some related work I did not finally did it because such changes results in more and more changes.

Reply:
Thank you. I really do think a MemBuf with stolen flag set would have been good enough to resolve all the issues both our viewpoints have with this logic. But since you have gone this far I wont request more work here.